### PR TITLE
Copy license to build output directory.

### LINKED ISSUE #2945: License text missing from chosen-package
While you include https://github.com/harvesthq/chosen/blob/master/LICENSE.md, it is missing from https://github.com/harvesthq/chosen-package

In the MIT license itself, it states...

> The above copyright notice and this permission notice shall be included in all
> copies or substantial portions of the Software.

You are making it MUCH harder for downstream users of this code to meet the licensing requirements when you don't include the LICENSE.md for Bower and NPM builds.

### PR DESCRIPTION
**Pull request by adunkman**
@harvesthq/chosen-developers to fix #2945.

This adds a step to the `build` grunt task which copies `LICENSE.md` to the output directory… which should theoretically include it in the final build? I’m not 100% sure, because build process is a little 🌁. 

Also, adds a `package-lock.json`, since it seems like we haven’t touched the dependencies in a while, and that’s a thing now.

bitmoji

### CODE REVIEW

**Comment by stof** [comment]
Should we also update the package.json of the built package, to include the license file in the npm archive ?

File: Gruntfile.coffee [hunk 1]
```diff
@@ -34,6 +34,12 @@ This file is generated by `grunt build`, do not edit it by hand.
         src: ['public/chosen.css']
         dest: 'public/chosen.css'
 
+    copy:
+      main:
+        files: [
+          {src: ['LICENSE.md'], dest: 'public/'}
```
[the remark this answers is not in the document]
Comment (adunkman) [reply Gruntfile.coffee]: Nice, I didn’t catch that syntax in the docs. I’ll update momentarily!

**Comment by adunkman** [comment]
LICENSE (with any capitalization or extension) is [automatically included in packages by npm](https://docs.npmjs.com/files/package.json#files) — sorry, I should have mentioned that in the description (I had to look it up!):

> Certain files are always included, regardless of settings:
> - LICENSE / LICENCE
> 
> README, CHANGES, LICENSE & NOTICE can have any case and extension.

**Comment by stof** [comment]
then it is fine

**Comment by adunkman** [comment]
Force-pushed an updated commit with the Gruntfile modifications above. Thanks for the review!

**Pull request merged**